lib/renderer.js

```diff
@@ -1164,10 +1183,12 @@
         // 4. The "include-ancestor-masks" config option is NOT set
         // 5. None of the layers has zero bounds. Sometimes they aren't computed and set to all 0's
         // 6. The layer is clipped, in which case layer.bounds is the clipped size and not the layer size
-        var hasComplexTransform = (component.hasOwnProperty("scale") && component.scale % 1 !== 0) ||
-                component.hasOwnProperty("width") || component.hasOwnProperty("height"),
+        var layer = component.layer,
+            hasComplexTransform = layer &&
```

Comment:
It's basically, I think the bug is in _getSettingsWithApproximateBounds instead. This is basically forcing all layer comps into the slow bounds calculation case. In the code there's this clause:

https://github.com/adobe-photoshop/generator-assets/blob/ea884529c40f5b3752273fad4a88be17f0a9b207/lib/renderer.js#L862

The bounds for the layer comp should be the same as the bounds for the regular document. What is the output you're seeing from both cases?